core/domain/user_services_test.py

```diff
@@ -129,3 +133,239 @@ def test_get_user_id_from_username(self):
         # Return None for usernames which don't exist.
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
+
+
+class UpdateContributionMsecTests(test_utils.GenericTestBase):
+    """Test whether contribution date changes with publication of
+    exploration/collection and update of already published exploration/collection
+    """
+
+    EXP_ID = 'test_exp'
+    COL_ID = 'test_col'
+    COLLECTION_TITLE = 'title'
+    COLLECTION_CATEGORY = 'category'
+    COLLECTION_OBJECTIVE = 'objective'
+
+    def test_contribution_msec_updates_on_published_explorations(self):
```

Comment:
Maybe add another test for the case where the contributors to the exploration are a strict subset of those listed in owner_ids and editor_ids.